Authors:  Cecil Curry <leycec@gmail.com>
Upstream: no
Synopsis: Conditionally generate Doxygenated documentation. (Actually, disable
Doxygen documentation generation, at the moment. It currently fails with error.)
--- old/CMakeLists.txt	2010-06-22 14:26:09.000000000 +0000
+++ new/CMakeLists.txt	2010-07-02 23:39:16.311981830 +0000
@@ -210,30 +210,14 @@
 ADD_SUBDIRECTORY(plugins)
 ADD_SUBDIRECTORY(externalplugins)
 ADD_SUBDIRECTORY(textures)
-FIND_PACKAGE(Doxygen QUIET)
-if (DOXYGEN_FOUND)
-  OPTION(GENERATE_DOC "Do you want to generate the documentation ? [OFF|ON]")
-  if (GENERATE_DOC)
-    ADD_SUBDIRECTORY(docs)
-    SET(TULIP_LIB_DESC "Tulip Graph Library")
-    SET(TULIP_LIB_NAME "tulip")
-    CONFIGURE_FILE("docs/doxygen/tulip-lib.doxygen.in" "docs/doxygen/tulip-lib.doxygen")
-    SET(TULIP_LIB_DESC "Tulip Open GL Library")
-    SET(TULIP_LIB_NAME "tulip-ogl")
-    CONFIGURE_FILE("docs/doxygen/tulip-lib.doxygen.in" "docs/doxygen/tulip-ogl-lib.doxygen")
-    SET(TULIP_LIB_DESC "Tulip Qt Library")
-    SET(TULIP_LIB_NAME "tulip-qt")
-    CONFIGURE_FILE("docs/doxygen/tulip-lib.doxygen.in" "docs/doxygen/tulip-qt-lib.doxygen")
-    EXEC_PROGRAM(date ARGS "+%Y" OUTPUT_VARIABLE THIS_YEAR)
-    CONFIGURE_FILE("docs/doxygen/tulip.html.in" "docs/doxygen/tulip.html")
-    CONFIGURE_FILE("docs/doxygen/header.html.in" "docs/doxygen/header.html")
-    CONFIGURE_FILE("docs/doxygen/footer.html.in" "docs/doxygen/footer.html")
-    CONFIGURE_FILE("docs/doxygen/header-allPlugins.html.in" "docs/doxygen/header-allPlugins.html")
-    CONFIGURE_FILE("docs/doxygen/tulip-plugins.doxygen.in" "docs/doxygen/tulip-plugins.doxygen")    
-  endif(GENERATE_DOC)
-else(DOXYGEN_FOUND)
-MESSAGE("Doxygen not found: unable to generate documentation")
-endif(DOXYGEN_FOUND)
+OPTION(ENABLE_DOXYGEN "Enable installation of Doxygen-generated documentation" OFF)
+IF(ENABLE_DOXYGEN)
+  FIND_PACKAGE(Doxygen QUIET)
+  if (DOXYGEN_FOUND)
+  else(DOXYGEN_FOUND)
+  MESSAGE("Doxygen not found: unable to generate documentation")
+  endif(DOXYGEN_FOUND)
+ENDIF(ENABLE_DOXYGEN)
 
 ## -----------------------------------------------------------------------------------------------
 ## Set var to create tulip-config script
